File: LocalParks/LocalParks/ClientApp/app/data/order.ts
import * as _ from "lodash";
import { OrderItem } from "./orderItem"

export class Order {
    orderId: number;
    dateCreated: Date;
    orderNumber: string;
    items: Array<OrderItem> = new Array<OrderItem>();

    get total(): number {
        return _.sum(_.map(this.items,
            i => i.unitPrice * i.quantity
        ));
    };
}

File: LocalParks.Core/Domain/Shop/Order.cs
using LocalParks.Core.Domain.User;
using System;
using System.Collections.Generic;

namespace LocalParks.Core.Domain.Shop
{
    public class Order
    {
        public int OrderId { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DeliveryDate { get; set; }
        public string OrderNumber { get; set; }
        public ICollection<OrderItem> Items { get; set; }
        public LocalParksUser User { get; set; }
    }
}